samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/SubmitToGradingActionListener.java

```diff
@@ -814,18 +814,43 @@ private void prepareItemGradingPerItem(ActionEvent ae, DeliveryBean delivery,
 			answerModified = false;
 			for (int m = 0; m < grading.size(); m++) {
 				ItemGradingData itemgrading = grading.get(m);
-				if (itemgrading != null) {
-					log.debug("\n:ItemId>>itemTextId>>answerId "+ itemgrading.getPublishedItemId()+itemgrading.getPublishedItemTextId()+itemgrading.getPublishedAnswerId()+"\n");	    
-
-					itemgrading.setAgentId(AgentFacade.getAgentString());
-					itemgrading.setSubmittedDate(new Date());
-					if (itemgrading.getRationale() != null && itemgrading.getRationale().length() > 0) {
-						itemgrading.setRationale(TextFormat.convertPlaintextToFormattedTextNoHighUnicode(log, itemgrading.getRationale()));
+				if ((itemgrading !=null && itemgrading.getItemGradingId() == null) || itemgrading.getItemGradingId().intValue() <= 0) {
```

Comment:
Findbugs mentioned this as a possible NPE because the null check isn't preventing the || case from being executed. Same as below an itemgrading be null? Might need another check.